src/order/sup_indep.lean

```diff
@@ -22,74 +22,85 @@ In distributive lattices, this is equivalent to being pairwise disjoint.
 variables {α β ι ι' : Type*}
 
 namespace finset
-variables [lattice α] [order_bot α] [decidable_eq ι] [decidable_eq ι']
+section lattice
+variables [lattice α] [order_bot α]
 
 /-- Supremum independence of finite sets. -/
-def sup_indep (s : finset ι) (f : ι → α) : Prop := ∀ ⦃a⦄, a ∈ s → disjoint (f a) ((s.erase a).sup f)
+def sup_indep (s : finset ι) (f : ι → α) : Prop :=
+∀ ⦃t⦄, t ⊆ s → ∀ ⦃i⦄, i ∈ s → i ∉ t → disjoint (f i) (t.sup f)
```

Comment:
I think this deserves a remark in the doc-comment saying that you avoid the more obvious `∀ i ∈ s, disjoint (f i) ((s.erase i).sup f)` definition as it needs decidable equality.